api/envoy/extensions/filters/http/router/v3/router.proto

```diff
@@ -134,4 +135,25 @@ message Router {
   // upstream HTTP filters will count as a final response if hedging is configured.
   // [#extension-category: envoy.filters.http.upstream]
   repeated network.http_connection_manager.v3.HttpFilter upstream_http_filters = 8;
+
+  message DestinationMonitor {
+    string monitor_name = 1 [(validate.rules).string = {min_len: 1}];
+    ;
+  };
+
+  message HttpEvents {
+    repeated DestinationMonitor send_to = 1 [(validate.rules).repeated = {min_items: 1}];
+    config.common.matcher.v3.MatchPredicate match = 2 [(validate.rules).message = {required: true}];
+  };
+
+  message LocallyOriginatedEvents {
+    repeated DestinationMonitor send_to = 1 [(validate.rules).repeated = {min_items: 1}];
+  };
+
+  message OutlierDetection {
+    repeated HttpEvents http_events = 1;
+    LocallyOriginatedEvents locally_originated_events = 2;
+  };
+
+  OutlierDetection outlier_detection = 10;
```

Comment:
IMO, we may needn't this for now? What we need to do is to unify all `putHttpResponseCode` to `putResult` (may could be done at separated PR). Then, at the upstream outlier extension, we will accept all reports from the `putResult` calling.

Reply:
Please see https://github.com/envoyproxy/envoy/pull/34154#discussion_r2085302601. Yes, we need a matcher at the router to determine what is error. `putHttpResponseCode` and `putResult` talk only using Http codes and cannot be used with "new" extensions. If we agree on the code logic, I will update docs to explain how "new" extensions work in comparison to "old" canned outlier detection mechanism.

Reply:
I think `putResult` will take a `Result` enum and the extension could also use the `Result` enum to indicate whether there is failure or success?

Reply:
We could unify all reporting to putResult first, see https://github.com/envoyproxy/envoy/pull/39440

Reply:
`Result` is not needed anymore. As I explained above it was necessary evil to span two scenarios, which are not needed in "new" extensions. All we need now is error/no-error. 
The goal of the latest changes is to make it protocol independent, so all logic of interpreting, mapping, etc is moved to router (and in the future to redis/dubbo, etc) and router sends error/no-error to outlier.
`putResult`operates on httpCodes which indicates it is sort of protocol-dependent.

I relly hope we can remove "old" outlier, `Result` and `putResult` in the future.